src/modules/exposure-period/dto/get-exposure-period-query.dto.ts

```diff
@@ -12,6 +13,6 @@ export class GetExposurePeriodQueryDto {
 
   @IsString()
   @ApiProperty({ example: 'EW', description: 'Two products are accepted: EW and BS' })
-  @Matches(/^(EW|BS)$/)
+  @IsEnum(ENUMS.PRODUCTS)
```

Comment:
does this work? the `@Matches` validator may not translate directly into `@IsEnum`

Reply:
Yep this does work.

Reply:
ENUM.PRODUCTS contain object { QueryParamProductsEnum: { EW: 'EW', BS: 'BS' } } . 
Enum has to be spread for ENUMS.PRODUCTS  or referenced using ENUMS.PRODUCTS.QueryParamProductsEnum

Reply:
Have rectified, accompanied with enhanced test coverage.